### DISCUSSION TITLE
DuckDB CLI syntax highlighting and autocomplete not working on Windows

### DISCUSSION THREAD

**Discussion by sullivanrs** [duckdb/duckdb#9271] [Ideas]
### What happens?

DuckDB CLI does not support syntax highlighting or autocomplete on Windows x64

### To Reproduce

When running DuckDB CLI from Powershell prompt, there is no syntax highlighting of SQL keywords (SELECT etc.) and no ability to autocomplete (even when autocomplete extension installed and loaded).

When run command ".highlight on" command to try to enable syntax highlighting the following error message is returned 

"Error: unknown command or invalid arguments:  "highlight". Enter ".help" for help"
 

### OS:

Windows x64

### DuckDB Version:

0.9

### DuckDB Client:

CLI

### Full Name:

Robert Sullivan

### Affiliation:

Private individual

### Have you tried this on the latest `main` branch?

I have tested with a main build

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] Yes, I have

**Comment by l1t1** [comment]
it never works since I use it

**Comment by ghdev75** [comment]
> it never works since I use it

Hi, I'm new to duckb there are my 2cents with testing with 
- Window 11 Version 10.0.22621 Build 22621
- Windows Terminal Version : 1.17.11461.0

duckdb 0.81 or 0.90 

### Via PowerShell 7.3.7 or native Duckdb shell

- Autocomplete extension/feature does not work (tab key is not handled by the shell ?)

- **.keyword** [color] and **.constant** [color] commands return an error

- **.keywordcode** 0 (0 for example since the **.help** command did not provide any hint how to use this command 'code' = 'ansi code ?'
> this cause the command prompt to be trashed with previous entries from history command or with the **.keywordcode** command parameters.
> Ditto with the **constantcode** command

### Ubuntu 20.04@WSL2 

With the same Windows Terminal, WSL ubuntu 20.04 with the latests updates.

- **autocomplete** extension, **.keyword** and **.constant** works just fine.
- same issue with **.keywordcode** and **.constantcode** and the side effect on the entry prompt.
> more generally some (any) incorrect entry ( cli commands, not SQL ? ) cause phantom or previous entries to be mingled with current command currently typed.

- example  .prompt '⚫◗ ' with unicode works just fine but is partially highlighted as a constant.

This is puzzling since the new Windows Terminal is allegedly much more compliant with standard Terminal behaviour.
Besides it is weird that the same duckdb shell (windows / linux@wls2) behave quite differently regarding highlighting, autocomplete (tab handling) features in the same terminal (Windows Terminal) .

**Comment by lavoshnikov** [comment]
Unfortunately, can confirm - tested today, syntax highlighting does not work in DuckDB CLI on Windows 10 Enterprise 22H2. 
Ran in CLI  downloaded as duckdb_cli-windows-amd64, duckdb v0.10.1.

**Comment by gregorywaynepower** [comment]
DuckDB v1.0.0 1f98600c2c still does not have autocomplete and syntax highlighting.

**Comment by michaeldesu** [comment]
I agree with gregorywaynepower. In the meantime, https://harlequin.sh/ is a good workaround if you don't mind installing another tool.

**Reply by gregorywaynepower** [reply]
> I agree with gregorywaynepower. In the meantime, https://harlequin.sh/ is a good workaround if you don't mind installing another tool.

Thanks! I'll give it a go!
